code/datums/components/edible.dm

```diff
@@ -0,0 +1,271 @@
+/*!
+
+This component makes it possible to make things edible. What this means is that you can take a bite or force someone to take a bite (in the case of items).
+These items take a specific time to eat, and can do most of the things our original food items could.
+
+Behavior that's still missing from this component that original food items had that should either be put into seperate components or somewhere else:
+	Components:
+	Drying component (jerky etc)
+	Customizable component (custom pizzas etc)
+	Processable component (Slicing and cooking behavior essentialy, making it go from item A to B when conditions are met.)
+	Dunkable component (Dunking things into reagent containers to absorb a specific amount of reagents)
+
+	Misc:
+	Something for cakes (You can store things inside)
+
+*/
+/datum/component/edible
+	///Amount of reagents taken per bite
+	var/bite_consumption = 2
+	///Amount of bites taken so far
+	var/bitecount = 0
+	///Flags for food
+	var/food_flags = NONE
+	///Bitfield of the types of this food
+	var/foodtypes = NONE
+	///Size of the food's volume
+	var/volume = 50
+	///Amount of seconds it takes to eat this food
+	var/eat_time = 30
+	///List of tastes players can taste when they consume something, for example list("crisps" = 2, "salt" = 1)
+	var/list/tastes = list("nothing")
+	///Defines how much it lowers someones satiety (Need to eat, essentialy)
+	var/junkiness = 0
+	///Message to send when eating
+	var/list/eatverbs
+	///Callback to be ran for when you take a bite of something
+	var/datum/callback/after_eat
+	///Atom that owns this component
+	var/atom/owner

```

Comment:
Don't store the parent like this, cast it when you need it

Reply:
resolved?

Reply:
This var is still here so no